src/fragments/lib/datastore/flutter/data-access/delete-snippet.mdx

```diff
@@ -1,19 +1,22 @@
 Below, you query for an instance with an `id` of `123`, and then delete it, if found:
 
 ```dart
+import 'package:amplify_flutter/amplify_flutter.dart';
+
+import 'models/ModelProvider.dart';
+
 Future<void> deletePostsWithId() async {
-  final oldPosts = await Amplify.DataStore.query(
+  final postToDelete = (await Amplify.DataStore.query(
```

Comment:
`delete()` can accept a query predicate, right? Should we update this to use a predicate?

Reply:
No, the predicate used on `save` and `delete` is meant to be a check before committing the change, if the predicate doesn't matcher the data going to be manipulated, the operation will be rejected ([details](https://docs.amplify.aws/lib/datastore/sync/q/platform/flutter/#update-and-delete-with-predicate)). In amplify-flutter we don't have support of `delete a model type by query predicate`.